tests/test_connect.py

```diff
@@ -4,3 +4,8 @@
 def test_connect_api():
     puppetdb = pypuppetdb.connect()
     assert puppetdb.version == 'v4'
+
+
+def test_connect_with_statement():
+    with pypuppetdb.connect() as puppetdb:
+        assert puppetdb.version == 'v4'
```

Comment:
can we also assert that the connection is closed?

Reply:
that's a pretty pertient request! I was wondering the same when writing the test and ended up with the interrogation "how can I check this with the current tooling?"

our best bet, I think, would be to mock out either the disconnection method or the `requests.Session.close()` method and verify that the mock is getting called.

related to this yesterday I was thinking that for folks who don't want to/can't use the "with" statement, should we also provide a `.disconnect()` method so that it's possible to call it manually? If we stub out `requests.Session.close()` we can test that disconnection happens as expected in the non-"with" test case, too.

(I'm just not exactly sure how to use the `mock` and verify that the mocked out function is getting called... but I guess I can search around in the library's documentation)

Reply:
> should we also provide a .disconnect() method so that it's possible to call it manually?

that sounds like a good idea
but if you're not feeling up to do it now, we can do it in a next iteration